jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/ClientBuilder.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.jaxrs;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.palantir.remoting.ssl.SslConfiguration;
+import com.palantir.remoting.ssl.SslSocketFactories;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+import org.joda.time.Duration;
+
+public abstract class ClientBuilder {
+    private Optional<SSLSocketFactory> thisSslSocketFactory = Optional.absent();
+    private Optional<X509TrustManager> thisTrustManager = Optional.absent();
+    private Optional<Duration> thisConnectTimeout = Optional.absent();
+    private Optional<Duration> thisReadTimeout = Optional.absent();
+
+    final Optional<SSLSocketFactory> getThisSslSocketFactory() {
+        return thisSslSocketFactory;
+    }
+
+    final Optional<Duration> getThisConnectTimeout() {
+        return thisConnectTimeout;
+    }
+
+    final Optional<Duration> getThisReadTimeout() {
+        return thisReadTimeout;
+    }
+
+    /**
+     * Creates and returns a {@link T T client} from the builder configuration and parameters. Subsequent invocations
+     * usually return different object.
+     */
+    public abstract <T> T build(Class<T> serviceClass, String userAgent, List<String> uris);
```

Comment:
I like the separation between the JaxRs/Feign-specific stuff and the Builder that has no feign/okhttp dependency. What advantage do you see for making it an interface? IMO, the current model is simpler to understand&implement.